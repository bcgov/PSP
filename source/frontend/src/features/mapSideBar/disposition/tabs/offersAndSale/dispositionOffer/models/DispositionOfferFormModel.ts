import { Api_DispositionFileOffer } from '@/models/api/DispositionFile';
import { emptyStringtoNullable, toTypeCodeNullable } from '@/utils/formUtils';

export class DispositionOfferFormModel {
  dispositionOfferStatusTypeCode: string | null = null;
  offerName: string | null = null;
  offerDate: string | null = null;
  offerExpiryDate: string | null = null;
  offerAmount: number | null = null;
  offerNote: string | null = null;
  rowVersion: number | null = null;

  constructor(readonly id: number | null = null, readonly dispositionFileId: number) {
    this.id = id;
    this.dispositionFileId = dispositionFileId;
  }

  static fromApi(entity: Api_DispositionFileOffer): DispositionOfferFormModel {
    const model = new DispositionOfferFormModel(entity.id, entity.dispositionFileId);

    model.dispositionOfferStatusTypeCode = entity.dispositionOfferStatusTypeCode;
    model.offerName = entity.offerName;
    model.offerDate = entity.offerDate;
    model.offerExpiryDate = entity.offerExpiryDate;
    model.offerAmount = entity.offerAmount;
    model.offerNote = entity.offerNote;
    model.rowVersion = entity.rowVersion ?? null;

    return model;
  }

  toApi(): Api_DispositionFileOffer {
    return {
      id: this.id,
      dispositionFileId: this.dispositionFileId,
      dispositionOfferStatusTypeCode: emptyStringtoNullable(this.dispositionOfferStatusTypeCode),
      dispositionOfferStatusType: toTypeCodeNullable(this.dispositionOfferStatusTypeCode),
      offerName: emptyStringtoNullable(this.offerName),
      offerDate: emptyStringtoNullable(this.offerDate),
      offerExpiryDate: emptyStringtoNullable(this.offerExpiryDate),
      offerAmount: this.offerAmount,
      offerNote: emptyStringtoNullable(this.offerNote),
      rowVersion: this.rowVersion ?? 0,
    };
  }
}
