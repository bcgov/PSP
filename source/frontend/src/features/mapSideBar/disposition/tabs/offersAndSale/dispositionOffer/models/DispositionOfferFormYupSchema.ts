/* eslint-disable no-template-curly-in-string */
import * as yup from 'yup';

export const DispositionOfferFormYupSchema = yup.object().shape({
  dispositionOfferStatusTypeCode: yup.string().nullable(),
  offerName: yup
    .string()
    .nullable()
    .required('Offer Name is required')
    .max(1000, 'Offer Name must be at most ${max} characters'),
  offerDate: yup.string().nullable().required('Offer Date is required'),
  offerExpiryDate: yup.string().nullable(),
  offerAmount: yup.string().nullable().required('Offer Price is required'),
  offerNote: yup
    .string()
    .nullable()
    .max(2000, 'Offer Price comments must be at most ${max} characters'),
});
