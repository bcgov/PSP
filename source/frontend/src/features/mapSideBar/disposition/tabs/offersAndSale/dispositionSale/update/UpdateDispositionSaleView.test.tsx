import { createMemoryHistory } from 'history';

import Claims from '@/constants/claims';
import { DispositionSaleFormModel } from '@/features/mapSideBar/disposition/models/DispositionSaleFormModel';
import { mockLookups } from '@/mocks/lookups.mock';
import { lookupCodesSlice } from '@/store/slices/lookupCodes/lookupCodesSlice';
import { systemConstantsSlice } from '@/store/slices/systemConstants';
import { render, RenderOptions, waitFor, act } from '@/utils/test-utils';

import UpdateDispositionSaleView, {
  IUpdateDispositionSaleViewProps,
} from './UpdateDispostionSaleView';

const defaultInitialValues = new DispositionSaleFormModel(null, 1, 0);

const history = createMemoryHistory();

const onSave = vi.fn();
const onCancel = vi.fn();
const onSuccess = vi.fn();
const onError = vi.fn();

describe('Update Disposition Sale View', () => {
  const setup = async (
    renderOptions: RenderOptions & { props?: Partial<IUpdateDispositionSaleViewProps> } = {},
  ) => {
    // const ref = createRef<FormikProps<DispositionFormModel>>();
    const utils = render(
      <UpdateDispositionSaleView
        {...renderOptions.props}
        initialValues={renderOptions.props?.initialValues ?? defaultInitialValues}
        loading={renderOptions.props?.loading ?? false}
        onCancel={onCancel}
        onSave={onSave}
        onSuccess={onSuccess}
        onError={onError}
      />,
      {
        ...renderOptions,
        useMockAuthentication: true,
        claims: renderOptions?.claims ?? [Claims.DISPOSITION_EDIT],
        history: history,
        store: {
          [lookupCodesSlice.name]: { lookupCodes: mockLookups },
          [systemConstantsSlice.name]: { systemConstants: [{ name: 'GST', value: '5.0' }] },
        },
      },
    );

    return {
      ...utils,
      getCancelButton: () => utils.getByText(/Cancel/i),
      getFinalSaleAmountTextbox: () =>
        utils.container.querySelector(`input[name="finalSaleAmount"]`) as HTMLInputElement,
      getGSTCollectedAmountTextbox: () =>
        utils.container.querySelector(`input[name="gstCollectedAmount"]`) as HTMLInputElement,
    };
  };

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('renders as expected', async () => {
    const { asFragment } = await setup();
    await act(async () => {});
    const fragment = await waitFor(async () => asFragment());
    expect(fragment).toMatchSnapshot();
  });
});
