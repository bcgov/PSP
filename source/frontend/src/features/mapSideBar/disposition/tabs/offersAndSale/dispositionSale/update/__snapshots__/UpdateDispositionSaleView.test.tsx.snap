// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Disposition Sale View renders as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div />
  .c5.btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.4rem 1.2rem;
  min-height: 3rem;
  border: 0.2rem solid transparent;
  border-radius: 0.4rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  font-family: 'BCSans','Noto Sans',Verdana,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  cursor: pointer;
}

.c5.btn:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c5.btn:focus {
  outline-width: 0.4rem;
  outline-style: solid;
  outline-offset: 1px;
  box-shadow: none;
}

.c5.btn.btn-primary {
  border: none;
}

.c5.btn.btn-secondary {
  background: none;
}

.c5.btn.btn-info {
  border: none;
  background: none;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
}

.c5.btn.btn-info:hover,
.c5.btn.btn-info:active,
.c5.btn.btn-info:focus {
  background: none;
}

.c5.btn.btn-light {
  border: none;
}

.c5.btn.btn-dark {
  border: none;
}

.c5.btn.btn-link {
  font-size: 1.6rem;
  font-weight: 400;
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 2.5rem;
  line-height: 3rem;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-align: left;
  padding: 0;
}

.c5.btn.btn-link:hover,
.c5.btn.btn-link:active,
.c5.btn.btn-link:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border: none;
  background: none;
  box-shadow: none;
  outline: none;
}

.c5.btn.btn-link:disabled,
.c5.btn.btn-link.disabled {
  background: none;
  pointer-events: none;
}

.c5.btn:disabled,
.c5.btn:disabled:hover {
  box-shadow: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.65;
}

.c5.Button .Button__icon {
  margin-right: 1.6rem;
}

.c5.Button--icon-only:focus {
  outline: none;
}

.c5.Button--icon-only .Button__icon {
  margin-right: 0;
}

.c8 .react-datepicker__calendar-icon {
  width: 3rem;
  height: 3rem;
  margin-top: 0.5rem;
  right: 0;
  pointer-events: none;
}

.c8 .react-datepicker__view-calendar-icon input {
  padding: 0.6rem 1rem 0.5rem 0.6rem;
}

.c8 .react-datepicker-wrapper {
  max-width: 16rem;
}

.c9.c9.form-control.is-valid {
  background-image: none;
}

.c9.c9.form-control.is-invalid {
  border-color: #d8292f !important;
}

.c13 {
  position: -webkit-sticky;
  position: sticky;
  padding-top: 2rem;
  bottom: 0;
  background: white;
  z-index: 10;
}

.c6 {
  background: none;
  position: relative;
  border-radius: 0.3rem;
  padding: 0.6rem;
  padding-right: 2.1rem;
  border: solid 0.1rem;
}

.c6.is-invalid {
  border: solid 0.1rem;
}

.c7.c7.btn {
  position: absolute;
  top: calc(50% - 1.4rem);
  right: 0.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  line-height: unset;
}

.c7.c7.btn .text {
  display: none;
}

.c7.c7.btn:hover,
.c7.c7.btn:active,
.c7.c7.btn:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: unset;
}

.c10 .react-datepicker__calendar-icon {
  width: 3rem;
  height: 3rem;
  margin-top: 0.5rem;
  right: 0;
  pointer-events: none;
}

.c10 .react-datepicker__view-calendar-icon input {
  padding: 0.6rem 1rem 0.5rem 0.6rem;
}

.c10 .react-datepicker-wrapper {
  max-width: 16rem;
}

.c10 .react-datepicker-year-header {
  font-size: 1.6rem;
  padding: 0.55rem 0;
}

.c10 .react-datepicker__navigation {
  top: 0.39rem;
}

.c11.c11.form-control.is-valid {
  background-image: none;
}

.c11.c11.form-control.is-invalid {
  border-color: #d8292f !important;
}

.c2 {
  font-weight: bold;
  border-bottom: 0.2rem solid;
  margin-bottom: 2rem;
}

.c1 {
  margin: 1.5rem;
  padding: 1.5rem;
  background-color: white;
  text-align: left;
  border-radius: 0.5rem;
}

.c4.required::before {
  content: '*';
  position: absolute;
  top: 0.75rem;
  left: 0rem;
}

.c3 {
  font-weight: bold;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
  height: 100%;
  overflow-y: auto;
  padding-bottom: 1rem;
}

.c12 {
  margin-right: 1rem;
  padding-bottom: 1rem;
  z-index: 0;
}

<div
    class="c0"
  >
    <div
      class=""
    >
      <div
        class="c1 form-section"
      >
        <h2
          class="c2"
        >
          <div
            class="no-gutters row"
          >
            <div
              class="col"
            >
              Sales Details
            </div>
          </div>
        </h2>
        <div
          class="collapse show"
        >
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Purchaser name(s):
              </label>
            </div>
            <div
              class="c4 text-left col-7"
            >
              <button
                class="c5 Button mb-3 btn btn-link"
                data-testid="add-purchaser-button"
                type="button"
              >
                <div
                  class="Button__value"
                >
                  + Add another purchaser
                </div>
              </button>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Purchaser agent:
              </label>
            </div>
            <div
              class="c4 text-left col-6"
            >
              <div
                class="form-group"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="c6"
                    >
                      Select from contacts
                      <button
                        class="c5 Button c7 btn btn-link"
                        title="remove"
                        type="button"
                      >
                        <div
                          class="Button__value"
                        >
                          <svg
                            fill="currentColor"
                            height="2rem"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="2rem"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </div>
                      </button>
                    </div>
                    <div
                      class="d-none input form-group"
                    >
                      <input
                        class="form-control"
                        id="input-dispositionPurchaserAgent.contact.id"
                        name="dispositionPurchaserAgent.contact.id"
                        placeholder="Select from Contacts"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="pl-0 col-auto"
                  >
                    <button
                      class="c5 Button Button--icon-only px-2 btn btn-primary"
                      title="Select Contact"
                      type="button"
                    >
                      <div
                        class="Button__icon"
                      >
                        <svg
                          fill="currentColor"
                          height="20"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 448 512"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-228-32c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H118.4C106 384 96 375.4 96 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z"
                          />
                        </svg>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
              <div />
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Purchaser solicitor:
              </label>
            </div>
            <div
              class="c4 text-left col-6"
            >
              <div
                class="form-group"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="c6"
                    >
                      Select from contacts
                      <button
                        class="c5 Button c7 btn btn-link"
                        title="remove"
                        type="button"
                      >
                        <div
                          class="Button__value"
                        >
                          <svg
                            fill="currentColor"
                            height="2rem"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="2rem"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </div>
                      </button>
                    </div>
                    <div
                      class="d-none input form-group"
                    >
                      <input
                        class="form-control"
                        id="input-dispositionPurchaserSolicitor.contact.id"
                        name="dispositionPurchaserSolicitor.contact.id"
                        placeholder="Select from Contacts"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="pl-0 col-auto"
                  >
                    <button
                      class="c5 Button Button--icon-only px-2 btn btn-primary"
                      title="Select Contact"
                      type="button"
                    >
                      <div
                        class="Button__icon"
                      >
                        <svg
                          fill="currentColor"
                          height="20"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 448 512"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-228-32c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H118.4C106 384 96 375.4 96 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z"
                          />
                        </svg>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
              <div />
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Last condition removal date:
                <span
                  class="ml-2"
                >
                  <span
                    class="tooltip-icon"
                    data-testid="tooltip-icon-section-field-tooltip"
                    id="section-field-tooltip"
                  >
                    <svg
                      class="tooltip-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                      />
                    </svg>
                  </span>
                </span>
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="c8 form-group"
              >
                <div
                  class="react-datepicker-wrapper d-block"
                >
                  <div
                    class="react-datepicker__input-container react-datepicker__view-calendar-icon"
                  >
                    <svg
                      class="react-datepicker__calendar-icon"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                      />
                    </svg>
                    <input
                      autocomplete="off"
                      class="c9 form-control date-picker"
                      id="datepicker-finalConditionRemovalDate"
                      name="finalConditionRemovalDate"
                      placeholder="MMM DD, YYYY"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Sale completion date:
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="c8 form-group"
              >
                <div
                  class="react-datepicker-wrapper d-block"
                >
                  <div
                    class="react-datepicker__input-container react-datepicker__view-calendar-icon"
                  >
                    <svg
                      class="react-datepicker__calendar-icon"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                      />
                    </svg>
                    <input
                      autocomplete="off"
                      class="c9 form-control date-picker"
                      id="datepicker-saleCompletionDate"
                      name="saleCompletionDate"
                      placeholder="MMM DD, YYYY"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Fiscal year of sale:
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="c10 form-group"
              >
                <div
                  class="react-datepicker-wrapper d-block"
                >
                  <div
                    class="react-datepicker__input-container react-datepicker__view-calendar-icon"
                  >
                    <svg
                      class="react-datepicker__calendar-icon"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                      />
                    </svg>
                    <input
                      class="c11 form-control"
                      id="datepicker-saleFiscalYear"
                      name="saleFiscalYear"
                      placeholder="YYYY"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Final sale price ($):
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-tooltip-wrapper"
                >
                  <input
                    class="form-control input-number"
                    id="input-finalSaleAmount"
                    inputmode="numeric"
                    name="finalSaleAmount"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Realtor commission ($):
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-tooltip-wrapper"
                >
                  <input
                    class="form-control input-number"
                    id="input-realtorCommissionAmount"
                    inputmode="numeric"
                    name="realtorCommissionAmount"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                GST required:
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <select
                  class="form-control"
                  id="input-isGstRequired"
                  name="isGstRequired"
                >
                  <option
                    value="true"
                  >
                    Yes
                  </option>
                  <option
                    value="false"
                  >
                    No
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Net Book Value ($):
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-tooltip-wrapper"
                >
                  <input
                    class="form-control input-number"
                    id="input-netBookAmount"
                    inputmode="numeric"
                    name="netBookAmount"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Total cost of sales ($):
                <span
                  class="ml-2"
                >
                  <span
                    class="tooltip-icon"
                    data-testid="tooltip-icon-section-field-tooltip"
                    id="section-field-tooltip"
                  >
                    <svg
                      class="tooltip-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                      />
                    </svg>
                  </span>
                </span>
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-tooltip-wrapper"
                >
                  <input
                    class="form-control input-number"
                    id="input-totalCostAmount"
                    inputmode="numeric"
                    name="totalCostAmount"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Net proceeds before SPP cost ($):
                <span
                  class="ml-2"
                >
                  <span
                    class="tooltip-icon"
                    data-testid="tooltip-icon-section-field-tooltip"
                    id="section-field-tooltip"
                  >
                    <svg
                      class="tooltip-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                      />
                    </svg>
                  </span>
                </span>
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-tooltip-wrapper"
                >
                  <input
                    class="form-control input-number"
                    disabled=""
                    id="input-netProceedsBeforeSppAmount"
                    inputmode="numeric"
                    name="netProceedsBeforeSppAmount"
                    placeholder=""
                    type="text"
                    value="$0.00"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                SPP Amount ($):
                <span
                  class="ml-2"
                >
                  <span
                    class="tooltip-icon"
                    data-testid="tooltip-icon-section-field-tooltip"
                    id="section-field-tooltip"
                  >
                    <svg
                      class="tooltip-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                      />
                    </svg>
                  </span>
                </span>
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-tooltip-wrapper"
                >
                  <input
                    class="form-control input-number"
                    id="input-sppAmount"
                    inputmode="numeric"
                    name="sppAmount"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Net proceeds after SPP cost ($):
                <span
                  class="ml-2"
                >
                  <span
                    class="tooltip-icon"
                    data-testid="tooltip-icon-section-field-tooltip"
                    id="section-field-tooltip"
                  >
                    <svg
                      class="tooltip-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                      />
                    </svg>
                  </span>
                </span>
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-tooltip-wrapper"
                >
                  <input
                    class="form-control input-number"
                    disabled=""
                    id="input-netProceedsAfterSppAmount"
                    inputmode="numeric"
                    name="netProceedsAfterSppAmount"
                    placeholder=""
                    type="text"
                    value="$0.00"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-5"
            >
              <label
                class="c3"
              >
                Remediation cost ($):
              </label>
            </div>
            <div
              class="c4 text-left col-5"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-tooltip-wrapper"
                >
                  <input
                    class="form-control input-number"
                    id="input-remediationAmount"
                    inputmode="numeric"
                    name="remediationAmount"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c13 justify-content-end mt-auto no-gutters row"
      >
        <div
          class="pr-3 col-auto"
        />
        <div
          class="pr-4 col-auto"
        >
          <button
            class="c5 Button btn btn-secondary"
            type="button"
          >
            <div
              class="Button__value"
            >
              Cancel
            </div>
          </button>
        </div>
        <div
          class="col-auto"
        >
          <button
            class="c5 Button btn btn-primary"
            type="button"
          >
            <div
              class="Button__value"
            >
              Save
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
